amr-wind/equation_systems/icns/source_terms/RayleighDamping.H

```diff
@@ -0,0 +1,52 @@
+#ifndef RAYLEIGH_DAMPING_H
+#define RAYLEIGH_DAMPING_H
+
+#include "amr-wind/equation_systems/icns/MomentumSource.H"
+#include "amr-wind/core/SimTime.H"
+
+namespace amr_wind::pde::icns {
+
+/** Apply Rayleigh damping as a source terms
+ * \ingroup icns_src
+ *
+ * The implementation is based on a private communication I (G. Deskos)
+ * had with Matt Churchfield
+ */
+
+class RayleighDamping : public MomentumSource::Register<RayleighDamping>
+{
+public:
+    static std::string identifier() { return "RayleighDamping"; }
+
+    explicit RayleighDamping(const CFDSim& /*sim*/);
+
+    ~RayleighDamping() override;
+
+    void operator()(
+        const int lev,
+        const amrex::MFIter& mfi,
+        const amrex::Box& bx,
+        const FieldState fstate,
+        const amrex::Array4<amrex::Real>& src_term) const override;
+
+private:
+    const amrex::AmrCore& m_mesh;
+
+    const Field& m_velocity;
+
+    //! Reference velocity defined as an input
+    amrex::Vector<amrex::Real> m_ref_vel{{15.0, 0.0, 0.0}};
+
+    //! Full length of the damping layer
+    amrex::Real m_dRD{500.};
+
+    //! Damping length
+    amrex::Real m_dFull{300.};
```

Comment:
Minor comment -- I was confused the first time I read this because in the if statement in `RayleighDamping.cpp` below, I thought `dFull` was for the full length length of the damping layer, and `dRD` to only cover the region where the Rayleigh damping coefficient is 1.0.  But the way it's coded, you always want `m_dFull` < `m_dRD`

Reply:
The new way I've set it up is that dFull corresponds to "length_complete_damping", where coeff = 1, and dRD corresponds to "length_sloped_damping", where coeff goes from 1 to 0, and dRD + dFull = total damping length. So these two input quantities are independent.